includes/enrolment/class-sensei-enrolment-course-calculation-job.php

```diff
@@ -153,4 +188,144 @@ private function get_query_args( $course_enrolment ) {
 
 		return $user_args;
 	}
+
+	/**
+	 * Modify user query to add the user ID check.
+	 *
+	 * @access private
+	 *
+	 * @param WP_User_Query $user_query User query to modify.
+	 */
+	public function modify_user_query_add_user_id( WP_User_Query $user_query ) {
+		global $wpdb;
+
+		$user_query->query_where .= $wpdb->prepare( ' AND ID>%d', $this->get_last_user_id() );
+	}
+
+	/**
+	 * Get the option name for tracking the current job.
+	 *
+	 * @return string
+	 */
+	private function get_current_job_option_name() {
+		$invalidated_suffix = $this->invalidated_only ? 1 : 0;
+
+		return self::OPTION_TRACK_CURRENT_JOB_PREFIX . $this->course_id . '_' . $invalidated_suffix;
+	}
+
+	/**
+	 * Get the option name for tracking the last user ID.
+	 *
+	 * @return string
+	 */
+	private function get_last_user_id_option_name() {
+		$job_id = $this->get_job_id();
+
+		if ( ! $job_id ) {
+			return false;
+		}
+
+		return self::OPTION_TRACK_LAST_USER_ID_PREFIX . $job_id;
+	}
+
+	/**
+	 * Set the last calculated user ID.
+	 *
+	 * @param int $user_id User ID.
+	 */
+	public function set_last_user_id( $user_id ) {
+		$current_user_option_name = $this->get_last_user_id_option_name();
+
+		if ( ! $current_user_option_name ) {
+			return;
+		}
+
+		update_option( $current_user_option_name, (int) $user_id, false );
+	}
+
+	/**
+	 * Get the last user ID that was calculated.
+	 *
+	 * @return int
+	 */
+	public function get_last_user_id() {
+		$current_user_option_name = $this->get_last_user_id_option_name();
+
+		if ( ! $current_user_option_name ) {
+			return 0;
+		}
+
+		return (int) get_option( $current_user_option_name, 0 );
+	}
+
+	/**
+	 * Get the current job ID.
+	 *
+	 * @return string|false
+	 */
+	public function get_current_job_id() {
+		global $wpdb;
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching -- Avoiding cache issues with multiple jobs running.
+		$row = $wpdb->get_row( $wpdb->prepare( "SELECT option_value FROM {$wpdb->options} WHERE option_name = %s LIMIT 1", $this->get_current_job_option_name() ) );
+
+		if ( is_object( $row ) && ! empty( $row->option_value ) ) {
+			return sanitize_text_field( $row->option_value );
+		}
+
+		return false;
+	}
+
+	/**
+	 * Get the job ID for this job.
+	 *
+	 * @return string
+	 */
+	public function get_job_id() {
+		return $this->job_id;
+	}
+
+	/**
+	 * Set up job before it is scheduled for the first time.
+	 */
+	public function start() {
+		$this->job_id = md5( uniqid() );
+
+		update_option( $this->get_current_job_option_name(), $this->job_id, false );
+
+		$this->set_last_user_id( 0 );
+	}
+
+	/**
+	 * Resume the current job. Useful for fetching before cancelling.
+	 *
+	 * @return bool True if we were able to restore the current job.
+	 */
+	public function resume() {
```

Comment:
Maybe rename this to `is_running`?

Reply:
This does more than a check for if it is running. It changes the state to so that it resumes the currently running job.

Reply:
Ah yes didn't read this correctly.